### PR TITLE
Add timestamp

### PR DESCRIPTION
**Pull request by cuteolaf**

### CODE REVIEW

File: pallets/proposals/src/impls.rs [hunk 1]
```diff
@@ -207,7 +210,7 @@ impl<T: Config> Pallet<T> {
                 .unwrap_or(&default_max_cap);
 
             ensure!(
-                max_cap == default_max_cap || max_cap >= new_contribution_value.clone(),
+                max_cap == default_max_cap || max_cap >= old_contribution.value.clone(),
```
Comment (f-gate) [inline pallets/proposals/src/impls.rs:213]: Looks like this should be new_contribution?
Any reason why its changed?

Comment (f-gate) [reply pallets/proposals/src/impls.rs:213]: Also no need to clone() see if you can borrow both with &. Cloning is a deep copy and is expensive.

File: pallets/proposals/src/impls.rs [hunk 2]
```diff
@@ -228,9 +231,15 @@ impl<T: Config> Pallet<T> {
...
 
-        project
-            .contributions
-            .insert(who.clone(), new_contribution_value.clone());
+        let timestamp = <pallet_timestamp::Pallet<T>>::get();
+
+        project.contributions.insert(
+            who.clone(),
+            Contribution {
+                value: old_contribution.value.saturating_add(value),
```
Comment (f-gate) [inline pallets/proposals/src/impls.rs:239]: nice defensive!

File: pallets/proposals/src/mock.rs [hunk 3]
```diff
@@ -273,4 +277,4 @@ pub fn build_test_externality() -> sp_io::TestExternalities {
     let mut ext = sp_io::TestExternalities::new(t);
     ext.execute_with(|| System::set_block_number(1));
     ext
-}
\ No newline at end of file
+}
```
Comment (f-gate) [inline pallets/proposals/src/mock.rs:280]: Maybe there could be a test to sanity check your code. To see if its working as you intend. Or not.

**Review by f-gate — COMMENTED** [review]
Looks good, very minor comments.

File: pallets/proposals/src/impls.rs (hunk 2 above)
Comment (samelamin) [reply pallets/proposals/src/impls.rs:239]: +1

File: pallets/proposals/src/impls.rs (hunk 1 above)
Comment (samelamin) [reply pallets/proposals/src/impls.rs:213]: correct, it should be your total contribution i.e new contribution

**Comment by samelamin** [comment]
Echoing @f-gate it looks good but lets add a test

Also I  think because this is a pr to the rebase branch the tests didn't run because we do have a test against max cap contributions that should have failed with using `old_contribution`. Can you please verify if that is the case @cuteolaf ?

File: pallets/proposals/src/impls.rs (hunk 1 above)
Comment (cuteolaf) [reply pallets/proposals/src/impls.rs:213]: it's a mistake, let me fix it.

**Comment by cuteolaf** [comment]
fixed & pushed a commit.

**Comment by samelamin** [comment]
@cuteolaf im merging in but can you please add a pr to add a test that a new contribution gets assigned a timestamp value

**Pull request merged**